include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,353 @@
+/**
+ * @file
+ *
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_H
+#define ZEPHYR_INCLUDE_STEPPER_H
+
+/**
+ * @brief Stepper Motor Controller Interface
+ * @defgroup stepper_interface Stepper Motor Controller Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor micro step resolution options
+ */
+enum micro_step_resolution {
+	STEPPER_FULL_STEP,
+	STEPPER_MICRO_STEP_2,
+	STEPPER_MICRO_STEP_4,
+	STEPPER_MICRO_STEP_8,
+	STEPPER_MICRO_STEP_16,
+	STEPPER_MICRO_STEP_32,
+	STEPPER_MICRO_STEP_64,
+	STEPPER_MICRO_STEP_128,
+	STEPPER_MICRO_STEP_256,
+};
+
+enum stepper_direction {
+	STEPPER_DIRECTION_POSITIVE,
+	STEPPER_DIRECTION_NEGATIVE,
+};
+
+/**
```

Comment:
these all should be within a INTERNAL_HIDDEN block

Reply:
done. Thanks to @faxe1008